procurement_purchase_no_grouping/models/stock_rule.py

```diff
@@ -24,11 +23,18 @@ def _run_buy(self, product_id, product_qty, product_uom, location_id,
     def _make_po_get_domain(self, values, partner):
         domain = super()._make_po_get_domain(values, partner)
         if self.env.context.get('grouping', 'standard') == 'product_category':
-            if values.get("product_id"):
-                product = self.env["product.product"].browse(values["product_id"])
-                domain += (
-                    ("order_line.product_id.categ_id", "=", product.categ_id.id),
+            if values.get("supplier"):
+                product = values["supplier"].product_id
+                purchase = self.env["purchase.order"].search(
+                    [
+                        ("order_line.product_id.categ_id", "!=", product.categ_id.id),
+                        ("partner_id", "=", partner.id),
+                        ("date_planned", ">=", values['date_planned']),
+                        ("state", "=", 'draft'),
+                    ], limit=1
                 )
+                if purchase:
+                    domain += (("id", "!=", purchase.id),)
```

Comment:
Would it work fine this way?
```suggestion
                domain += (
                        ("order_line.product_id.categ_id", "==", product.categ_id.id),
                        ("partner_id", "=", partner.id),
                        ("date_planned", ">=", values['date_planned']),
                        ("state", "=", 'draft'),
                )
```

Reply:
Thanks but with your suggested changes don't work because only need to add domain to NOT add line in other purchase order, need to separate according to category.

Reply:
Fixed with a simplified domain, but using other value (not `product_id`, but `supplier`).